tests/functional/context_methods/test_builtin_functions.py

```diff
@@ -0,0 +1,55 @@
+import pytest
+
+from dbt.tests.util import run_dbt, write_file
+from dbt.exceptions import CompilationException
+
+
+tests__set_sql = """
+{% set set_result = set([1, 2, 2, 3, "foo", False]) %}
+{% set try_set_result = try_set([1, 2, 2, 3, "foo", False]) %}
+
+{% set simple_set = (set_result == try_set_result == set((1, 2, 3, "foo", False))) %}
```

Comment:
This check is a bit off since you're using the new method to check the new method.  

Instead of using tests I would use macros that log the values of `set_result` and`try_set_result`.  Something like:

```sql
{% macro validate_set() %}
    {% set set_result = set([1, 2, 2, 3, 'foo', False]) %}
    {{ log("set_result: " ~ set_result) }}
    {% set try_set_result = try_set([1, 2, 2, 3, 'foo', False]) %}
    {{ log("try_set_result: " ~ try_set_result) }}
{% endmacro %}
```

Then in the test you can capture the logs from `run-operation` with the `--debug` flag for the macro and assert the logs show up:

```python
def test_builtin_set_function(self, project):
        _, log_output = run_dbt_and_capture(["--debug", "run-operation", "validate_set"])

        expected_set = {False, 1, 2, 3, 'foo'}
        assert f"set_result: {expected_set}" in log_output
        assert f"try_set_result: {expected_set}" in log_output
```

Also do the same for zip for consistency.